source/hwIo.py

```diff
@@ -304,34 +323,42 @@ def getFeature(self, reportId):
 			log.debug("Get feature: %r" % buf.raw)
 		return buf.raw
 
-	def setFeature(self, report):
+	def setFeature(self, report: bytes) -> None:
 		"""Send a feature report to this device.
 		@param report: The report, including its id.
-		@type report: str
+		@type report: bytes
 		"""
-		length = len(report)
-		buf = ctypes.create_string_buffer(length)
-		buf.raw = report
+		buf = ctypes.create_string_buffer(report)

```

Comment:
I think this buffer is no 1 byte longer than expected, as a string buffer will probably contain a terminating character.

Reply:
Good catch, I can pass the size along with it to stop this.